explorer/src/Pos/Explorer/TestUtil.hs

```diff
@@ -0,0 +1,240 @@
+                                                    SlotLeaders, StakeholderId,
+                                                    difficultyL)
+import           Pos.Crypto                        (SecretKey)
+import           Pos.Delegation                    (DlgPayload, DlgUndo, ProxySKBlockInfo)
+import           Pos.Ssc.Core                      (SscPayload, defaultSscPayload)
+import           Pos.Txp.Core                      (TxAux)
+import           Pos.Update.Configuration          (HasUpdateConfiguration)
+import           Pos.Update.Core                   (UpdatePayload (..))
+
+
+----------------------------------------------------------------
+-- Arbitrary and Show instances
+----------------------------------------------------------------
+
+-- I used the build function since I suspect that it's safe (even in tests).
+instance HasConfiguration => Prelude.Show SlogUndo where
+    show = show . build
+
+instance Prelude.Show DlgUndo where
+    show = show . build
+
+instance HasConfiguration => Prelude.Show Undo where
+    show = show . build
+
+instance Arbitrary SlogUndo where
+    arbitrary = genericArbitrary
+
+instance HasConfiguration => Arbitrary DlgUndo where
+    arbitrary = genericArbitrary
+
+instance HasConfiguration => Arbitrary Undo where
+    arbitrary = genericArbitrary
+
+----------------------------------------------------------------
+-- Utility
+-- TODO(ks): Extract this in some common PureBlockTest module in src/?
+----------------------------------------------------------------
+
+basicBlockGenericUnsafe
+    :: (HasConfiguration, HasUpdateConfiguration)
+    => BlockHeader
+    -> SecretKey
+    -> SlotId
+    -> Block
+basicBlockGenericUnsafe prevHeader sk slotId = case (basicBlock prevHeader sk slotId) of
+    Left e      -> error e
+    Right block -> Right block
+
+basicBlock
+    :: (HasConfiguration, HasUpdateConfiguration)
+    => BlockHeader
+    -> SecretKey
+    -> SlotId
+    -> Either Text MainBlock
+basicBlock prevHeader sk slotId =
+    producePureBlock infLimit prevHeader [] Nothing slotId def (defGTP slotId) def sk
+
+emptyBlk
+    :: (HasConfiguration, HasUpdateConfiguration, Testable p)
+    => (Either Text MainBlock -> p)
+    -> Property
+emptyBlk testableBlock =
+    forAll arbitrary $ \(sk, prevHeader, slotId) ->
+    testableBlock
+        $ producePureBlock infLimit prevHeader [] Nothing slotId def (defGTP slotId) def sk
+
+defGTP :: HasConfiguration => SlotId -> SscPayload
+defGTP sId = defaultSscPayload $ siSlot sId
+
+infLimit :: Byte
+infLimit = convertUnit @Gigabyte @Byte 1
+
+producePureBlock
+    :: (HasConfiguration, HasUpdateConfiguration)
+    => Byte
+    -> BlockHeader
+    -> [TxAux]
+    -> ProxySKBlockInfo
+    -> SlotId
+    -> DlgPayload
+    -> SscPayload
+    -> UpdatePayload
+    -> SecretKey
+    -> Either Text MainBlock
+producePureBlock limit prev txs psk slot dlgPay sscPay usPay sk =
```

Comment:
Wow - we have 9 parameters here. This function seems to be a good candidate to win a :medal_sports: in our sports category `heavy-weight-parameter-function` (or maybe:1st_place_medal:)  :smile:

Reply:
Actually, this is an existing function that I copied, but forgot from where :thinking: 
It's this reason that I believe we should have a commonplace module for all the functions we use in tests.